libs/core/tests/sharing_tests.rs

```diff
@@ -329,6 +329,213 @@ fn write_document_in_rejected_shared_folder_in_rejected_share_folder() {
+}
+
+#[test]
+fn create_document_in_target_folder_by_sharee() {
+    let cores = vec![test_core_with_account(), test_core_with_account()];
+    let accounts = cores
+        .iter()
+        .map(|core| core.get_account().unwrap())
+        .collect::<Vec<_>>();
+
+    let folder1 = cores[0].create_at_path("/folder/").unwrap();
+
+    cores[0]
+        .share_file(folder1.id, &accounts[1].username, ShareMode::Write)
+        .unwrap();
+
+    cores[0].sync(None).unwrap();
+    cores[1].sync(None).unwrap();
+
+    cores[1].create_link_at_path("/link1", folder1.id).unwrap();
+    cores[1]
+        .create_file("document1", folder1.id, FileType::Document)
+        .unwrap();
+
+    cores[1].sync(None).unwrap();
+    cores[0].sync(None).unwrap();
+
+    assert::all_paths(&cores[1], &["/", "/link1/", "/link1/document1"]);
+    assert::all_paths(&cores[0], &["/", "/folder/", "/folder/document1"]);
+}
+
+#[test]
+fn create_document_in_target_folder_by_sharer() {
+    let cores = vec![test_core_with_account(), test_core_with_account()];
+    let accounts = cores
+        .iter()
+        .map(|core| core.get_account().unwrap())
+        .collect::<Vec<_>>();
+
+    let folder1 = cores[0].create_at_path("/folder/").unwrap();
+
+    cores[0]
+        .share_file(folder1.id, &accounts[1].username, ShareMode::Write)
+        .unwrap();
+
+    cores[0].sync(None).unwrap();
+    cores[1].sync(None).unwrap();
+
+    cores[1].create_link_at_path("/link1", folder1.id).unwrap();
+    cores[0]
+        .create_file("document1", folder1.id, FileType::Document)
+        .unwrap();
+
+    cores[0].sync(None).unwrap();
+    cores[1].sync(None).unwrap();
+
+    assert::all_paths(&cores[1], &["/", "/link1/", "/link1/document1"]);
+    assert::all_paths(&cores[0], &["/", "/folder/", "/folder/document1"]);
+}
+
+#[test]
+fn get_link_target_children_recursive_by_sharee() {
+    let cores = vec![test_core_with_account(), test_core_with_account()];
+    let accounts = cores
+        .iter()
+        .map(|core| core.get_account().unwrap())
+        .collect::<Vec<_>>();
+
+    let folder1 = cores[0].create_at_path("/folder/").unwrap();
+    cores[0].create_at_path("/folder/folder/").unwrap();
+    let document = cores[0].create_at_path("/folder/folder/document").unwrap();
+    cores[0]
+        .write_document(document.id, b"document content by sharer")
+        .unwrap();
+    cores[0]
+        .share_file(folder1.id, &accounts[1].username, ShareMode::Write)
+        .unwrap();
+
+    cores[0].sync(None).unwrap();
+    cores[1].sync(None).unwrap();
+
+    let link = cores[1].create_link_at_path("/link1", folder1.id).unwrap();
+
+    cores[0].sync(None).unwrap();
+    cores[1].sync(None).unwrap();
+
+    let result: Vec<Uuid> = cores[1]
+        .get_and_get_children_recursively(link.id)
+        .unwrap()
+        .iter()
+        .map(|f| f.id)
+        .collect();
+
+    assert::all_recursive_children_ids(&cores[1], folder1.id, &result);
```

Comment:
I think you want to assert something more specific here. Maybe a subsequent test has this, but you want to make sure:
+ `document` shows up in the recursive listing of `/`
+ `document` shows up in the recursive listing of `folder`

Reply:
You probably want a similar thing for non-recursive get children and list metadata with tweaks respectively.